homeassistant/components/zone/__init__.py

```diff
@@ -346,10 +340,23 @@ async def async_update_config(self, config: dict) -> None:
 
     @callback
     def _person_state_change_listener(self, evt: Event) -> None:
-        object_id = split_entity_id(self.entity_id)[1]
         person_entity_id = evt.data["entity_id"]
         cur_count = len(self._persons_in_zone)
-        if evt.data["new_state"] and evt.data["new_state"].state == object_id:
+        if (
+            evt.data["new_state"]
+            and None
+            not in (
+                (latitude := evt.data["new_state"].attributes.get(ATTR_LATITUDE)),
+                (longitude := evt.data["new_state"].attributes.get(ATTR_LONGITUDE)),
+                (accuracy := evt.data["new_state"].attributes.get(ATTR_GPS_ACCURACY)),
+            )
```

Comment:
Optional, mainly to match the suggested style in the other comment
```suggestion
            (state := evt.data["new_state"])
            and (latitude := state.attributes.get(ATTR_LATITUDE)) is not None
            and (longitude := state.attributes.get(ATTR_LONGITUDE)) is not None
            and (accuracy := state.attributes.get(ATTR_GPS_ACCURACY)) is not None
```